### DISCUSSION TITLE
On-demand ISR with hostname-rewrites not revalidating

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#38377] [Help] [answered]
## When trying to revalidate the page on demand by calling revalidate() it always fails

**Simple repository to reproduce the error**
https://github.com/ianmorais25/hostname-rewrites-isr

https://<your-site.com>/api/revalidate?secret=token
-->
Internal Server Error  
Error: Failed to revalidate /: Invalid response 404

**Comment by ijjk** [comment, marked as answer]
Hi, middleware is not executed for On-Demand ISR requests so the path passed to `res.revalidate()` needs to be the exact path that should be revalidated e.g. in the above reproduction it should be `res.revalidate('/_site/site-name')`.

**Comment by alicercedigital** [comment]
Why there is nothing about this on documentation?

**Reply by ijjk** [reply]
Hi, this is documented here https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation
